### PR TITLE
feat(argo-rollouts): Allow customizing additional RBAC rules for other providers

### PR DESCRIPTION
**Pull request by amcamina**
Allow customizing additional RBAC rules for other providers

In my case, we used the service mesh LinkerD. However, the helm chart doesn't give any option to add more rules to the `ClusterRole`.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

File: charts/argo-rollouts/templates/controller/clusterrole.yaml [hunk 1]
```diff
@@ -258,23 +258,26 @@ rules:
...
+  - networking.gloo.solo.io
   resources:
-    - routetables
+  - routetables
   verbs:
-    - '*'
+  - '*'
+{{- end }}
+{{- if .Values.providerRBAC.additionalRules }}
+{{ toYaml .Values.providerRBAC.additionalRules }}
```
Comment (yu-croco) [inline charts/argo-rollouts/templates/controller/clusterrole.yaml:280]: Maybe it's simpler ? 🙋 

```yaml
{{- with .Values.providerRBAC.additionalRules }}
{{ toYaml .  }}
{{- end }}

Comment (amcamina) [reply charts/argo-rollouts/templates/controller/clusterrole.yaml:280]: I'll like it. Thanks!

**Comment by yu-croco** [comment]
Hi @amcamina, thank you for your PR.

>Linting and Testing / chart-test (pull_request)

Maybe you should run `./scripts/helm-docs.sh` at this repo's root dir to format README.
Ref: https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation

> DCO

You can fix with the following.
Ref: https://github.com/argoproj/argo-helm/pull/2556/checks?check_run_id=22126507245
> To avoid having PRs blocked in the future, always include Signed-off-by: Author Name <authoremail@example.com> in every commit message. You can also do this automatically by using the -s flag (i.e., git commit -s).

**Review by yu-croco — APPROVED** [review]
Thank you for your contribution. LGTM.

**Comment by amcamina** [comment]
@jmeridth Could you review this PR?

**Review by jmeridth — APPROVED** [review]

**Pull request merged**